openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/SuggestionRepository.java

```diff
@@ -220,6 +238,70 @@ protected void acceptSuggestion(
     update(suggestion, user);
   }
 
+  protected void acceptSuggestionList(
+      List<Suggestion> suggestions, SecurityContext securityContext, Authorizer authorizer) {
+    String user = securityContext.getUserPrincipal().getName();
+
+    // Entity being updated
+    EntityInterface entity = null;
+    EntityRepository<?> repository = null;
+    String origJson = null;
+    SuggestionWorkflow suggestionWorkflow = null;
+    SuggestionType suggestionType = getTypeFromSuggestions(suggestions);
+
+    for (Suggestion suggestion : suggestions) {
+      MessageParser.EntityLink entityLink =
+          MessageParser.EntityLink.parse(suggestion.getEntityLink());
+
+      // Validate all suggestions indeed talk about the same entity
+      if (entity == null) {
+        // Initialize the Entity and the Repository
+        entity =
+            Entity.getEntity(
+                entityLink, suggestionType == SuggestionType.SuggestTagLabel ? "tags" : "", ALL);
+        repository = Entity.getEntityRepository(entityLink.getEntityType());
+        origJson = JsonUtils.pojoToJson(entity);
+        suggestionWorkflow = repository.getSuggestionWorkflow(repository, entity);
+      } else if (!entity.getFullyQualifiedName().equals(entityLink.getEntityFQN())) {
+        throw new SuggestionException("All suggestions must be for the same entity");
+      }
+      // update entity with the suggestion
+      entity = suggestionWorkflow.acceptSuggestion(suggestion, entity);
+    }
+
+    // Patch the entity with the updated suggestions
+    String updatedEntityJson = JsonUtils.pojoToJson(entity);
+    JsonPatch patch = JsonUtils.getJsonPatch(origJson, updatedEntityJson);
+
+    OperationContext operationContext = new OperationContext(repository.getEntityType(), patch);
```

Comment:
would be better to this do early in the method rather than going through parsing of suggestions.

Reply:
I might be missing something here. But the idea is that we go over all the suggestions and build a single patch from them all (instead of patching 1 by 1). So we can only build this once all suggestions have been reviewed no?